### PR TITLE
use decodeURIComponent for router.query.next during page authentication redirect

### PR DESCRIPTION
**Pull request by remjx**
I ran into an issue where my `Page.authenticate = { redirectTo: "/login" }` was not working as expected. The initial page was `http://localhost:3000/page-i-refreshed` and the redirect URL was `http://localhost:3000/login?next=%2Fpage-i-refreshed` (this is OK so far). But after I logged in, the URL changed to `http://localhost:3000/%2Fpage-i-refreshed` instead of back to `http://localhost:3000/page-i-refreshed` which generated a 404. 

### What are the changes and their implications?

Use `decodeURIComponent` on the `router.query.next` path before `router.push`

Update `blitz generate` template
Update default login example

### Checklist

- [ ] Changes covered by tests (tests added if needed)
- [x] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes (N/A)

### CODE REVIEW

**Comment by remjx** [comment]
I'm having trouble getting the tests to run, I made a post in help channel in Discord https://discord.com/channels/802917734999523368/814628067788062750/820540099795091468

File: examples/auth/app/auth/pages/login.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const LoginPage: BlitzPage = () => {
     <div>
       <LoginForm
         onSuccess={() => {
-          const next = (router.query.next as string) ?? "/"
+          const next = decodeURIComponent(router.query.next as string) ?? "/"
```
Comment (flybayer) [inline examples/auth/app/auth/pages/login.tsx:13]: Hey, thank you! 

The issue is that `decodeURIComponent(undefined) === "undefined"` and so this doesn't fall back to `"/"`.

So probably need to change like this:
```suggestion
          const next = router.query.next ? decodeURIComponent(router.query.next) : "/"
```

Comment (remjx) [reply examples/auth/app/auth/pages/login.tsx]: That makes sense. Updated.

**Comment by flybayer** [comment]
The tests are still failing with a type error. Probably you need to do `decodeURIComponent(router.query.next!)` or cast it like it was.

Make sure the tests pass. You can run `yarn jest` from `examples/auth/` to run these tests.

**Comment by flybayer** [comment]
Yay, thank you!!

**Pull request merged**